File: src/main/java/com/rokucraft/RokuBot/util/EmbedUtil.java
package com.rokucraft.RokuBot.util;

import com.rokucraft.RokuBot.Main;
import net.dv8tion.jda.api.EmbedBuilder;

import static com.rokucraft.RokuBot.Constants.*;

public class EmbedUtil {
    public static EmbedBuilder createInfoEmbed() {
        return new EmbedBuilder()
                .setColor(BLUE)
                .setThumbnail("https://raw.githubusercontent.com/twitter/twemoji/master/assets/72x72/2139.png");
    }

    public static EmbedBuilder createIssuesEmbed(String issueList) {
        return new EmbedBuilder()
                .setTitle("Open Issues", Main.defaultRepo.getHtmlUrl() + "/issues")
                .setDescription(issueList)
                .setThumbnail("https://cdn.discordapp.com/attachments/786216721065050112/787721554992824360/issue-opened72px.png")
                .setColor(ISSUE_OPEN_COLOR)
                .setFooter("");
    }

    public static EmbedBuilder createErrorEmbed() {
        return new EmbedBuilder()
                .setColor(RED);
    }
}

File: src/main/kotlin/com/rokucraft/rokubot/util/EmbedUtil.kt
package com.rokucraft.rokubot.util

import com.rokucraft.rokubot.util.EmbedUtil.createErrorEmbed
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.entities.MessageEmbed
import net.dv8tion.jda.api.interactions.callbacks.IReplyCallback

object EmbedUtil {
    fun createErrorEmbed(description: String): MessageEmbed {
        return EmbedBuilder()
            .setColor(ColorConstants.RED)
            .setDescription("❌ $description")
            .build()
    }
}

fun IReplyCallback.replyError(description: String) {
    replyEmbeds(createErrorEmbed(description)).setEphemeral(true).queue()
}
